API.md

```diff
@@ -1,64 +1,272 @@
+    - [Resetting `globalSetup`](#resetting-globalsetup)
+  - [Helpers](#helpers)
+    - [before()](#before)
+    - [after()](#after)
+    - [finally()](#finally)
+    - [afterJSON()](#afterjson)
+    - [only()](#only)
+    - [timeout(ms)](#timeoutms)
+    - [retry(count, backoff)](#retrycount-backoff)
+    - [baseUri(uri)](#baseuriuri)
+    - [waits(ms)](#waitsms)
+    - [exceptionHandler(function)](#exceptionhandlerfunction)
+  - [Inspectors](#inspectors)
+    - [inspect(cb)](#inspectcb)
+    - [inspectRequest(message)](#inspectrequestmessage)
+    - [inspectResponse(message)](#inspectresponsemessage)
+    - [inspectHeaders(message)](#inspectheadersmessage)
+    - [inspectJSON(message)](#inspectjsonmessage)
+    - [inspectBody(message)](#inspectbodymessage)
+    - [inspectStatus(message)](#inspectstatusmessage)
+    - [Send Raw JSON or POST Body](#send-raw-json-or-post-body)
+
+## The Basics
+
+Every frisby request begins with `create(..)` and ends with a `toss()`.
+
+### create(msg)
+
+- Types: `msg`: `string`
+- Default: `none`
+
+Used to create an instance of IcedFrisby that is used to send 1 request and receive the response.
+The `msg` is used to name the test when it's wrapped in mocha for execution at runtime.
+
+```javascript
+frisby.create('a test')
+    .get('http://example.com')
+    .expectStatus(200)
+    // any number of additional expect statements here
+    .toss();
+```
+
+### toss()
+
+Used to complete the list of commands and issue the request.
+
+```javascript
+frisby.create('a test')
+    .get('http://example.com')
+    .expectStatus(200)
+    // any number of additional expect statements here
+    .toss();
+```
+
+## Commands
+
+These are the commands you'll need to get IcedFrisby talking HTTP.
+
+An optional parameters object is accepted in all of these methods, but aren't _useful_ for every verb. All object parameters are optional.
+
+```javascript
+{
+  json: 'boolean',         //Whether this will be a JSON body. Overrides value set in globalSetup().
+  body: 'string'|'object', //The body to include in the outbound request. This overrides "data" if provided in method call.
+  mock: 'function',        //A mock runner to use. When not provided, uses "request" (i.e. does it for real).
+  form: 'boolean'          //Use the object in the body to create a form-encoded request.
+}
+```
+
+### get(uri, params)
+
+- Types: `uri`: `string`, `params`: `object` (optional)
+- Default: `none`
+
+Perform an HTTP GET on the specified URI.
+
+```javascript
+frisby.create('a test')
+    .get('http://example.com/login')
+    .expectStatus(200)
+    // any number of additional expect statements here
+    .toss();
+```
+
+### head(uri, params)
+
+Identical to [get](#geturi-params), using an HTTP HEAD request.
+
+### options(uri, params)
+
+Identical to [get](#geturi-params), using an HTTP OPTIONS request.
+
+### post(uri, data, params)
+
+- Types: `uri`: `string`, `data`:`object` `params`: `object` (optional)
+- Default: `none`
```

Comment:
I realize this is the old format. In the future it would probably be clearer to list these one at a time, and indicate the default for each, and whether they are required:

- `uri`: The URI string.
- `data`: The data to post. An object when `{ json: true }` or `{ form: true }`, or else a data string / buffer.
- `params`: An optional params object, as described above.

Reply:
I didn't like this format either, but kept consistent for this piece. 
If there's no love for it, I might change it now. 

Another alternative:

| Parameter | Description | Required |
| ----------- | ------------- | ----------- |
| uri | The URI string | Yes |
| data | The data to post. An object when { json: true } or { form: true }, or else a data string / buffer. | Yes |
| params | An optional params object, as described above. | No |

It's clean and clear, but is it heavyweight?

Reply:
I'm fine with that and feel like it's an improvement. You could also add a type column.

Reply:
Done. I decided against the Type column since most of them were strings. Then wrote String almost everywhere anyway. We can revisit this later if we feel it's clunky. Maybe solicit some external feedback somehow?